### DISCUSSION TITLE
Why `logger.error` instead of `raise`

### DISCUSSION THREAD

**Discussion by numpde** [idom-team/idom#910] [Problem]
In [_validate_child_key_integrity](https://github.com/idom-team/idom/blob/bd874072f0899dea72d9d1647d0c24c5e48dc724/src/idom/core/vdom.py#L293) you have `logger.error` but no `raises`, is that intentional? The severity is not clear and the source of the issue is hard to locate based on the output.

**Comment by rmorshea** [comment]
Now that I'm looking at this again, raising an exception saying "Expected children to be passed as a sequence (e.g. list or tuple) instead of a generator" might be better. Ultimately what this log is trying to explain is that, because you passed a generator instead of a sequence, IDOM was not able to validate whether all child elements from the generator have keys.

**Reply by rmorshea** [reply]
On the other hand this isn't a critical failure. We could also warn instead. If we do this we should log at the appropriate stack level using [this utility](https://github.com/idom-team/idom/blob/main/src/idom/_warnings.py).

**Reply by Archmonger** [reply]
Perhaps if we can't validate it we may as well reject it?

`IDOM does not support generators. Utilize lists instead.`

It's feels more JavaScripty to require lists.
